### DISCUSSION TITLE
Lesson 5: Arrays and structs - Why should we call index first and string next?

### DISCUSSION THREAD

**Discussion by ljb630** [Cyfrin/foundry-full-course-cu#1539] [Q&A] [answered]
Sample Code:

**listofbfs.push(person(_favoriteNumber, _name)); // Here we need to call first index, next string**

**Comment by mxnwafor** [comment]
Your question is not structured properly. Could you rephrase?

**Comment by EngrPips** [comment]
Hello @ljb630 , we would really appreciate if you provide more context about what you are asking. you can maybe check how to properly ask question by clicking this link  https://github.com/Cyfrin/foundry-full-course-f23/blob/main/how-to-ask-a-question.md

**Comment by ljb630** [comment]
Okay, let me explain what I am referring to. In the function below, we are calling _favoriteNumber and _name. I am unsure why we are calling _favoriteNumber before _name, as I tried calling _name before _favoriteNumber and it resulted in an error. Can you help me understand the reason for this order?

```

   // dynamic array
    person[] public listofbfs;

    function addbfs( string memory _name, uint256 _favoriteNumber) public {
        listofbfs.push(person(_favoriteNumber, _name))// Here we need to call first index, next string
    }
```

**Reply by ljb630** [reply]
@EngrPips @0xMitchy  Please check the above comment.

**Reply by EngrPips** [reply]
Hello @ljb630 , can you please provide  the `listofbfs` data structure, where you declare it. Lets start with you providing that then i point out stuffs why it has to be in that order.

**Reply by ljb630** [reply]
My bad :)

```
    struct person {
        uint256 Myfavnumber;
        string name;
    }
```

**Reply by EngrPips** [reply, marked as answer]
Look at the struct you defined here you declared `Myfavnumber` which is aa type of uint256 before `name` which is a type of `string` , so when you want to create a new struct which is what you are doing in the line of code below 

```javascript
listofbfs.push(person(_favoriteNumber, _name))// Here we need to call first index, next string
```
Particularly in this piece of code 

```javascript
person(_favoriteNumber, _name)
```

You are trying to create a new Person and to do that you have to pass in their favoriteNumber first before their name because that was the order in which you declared the Person struct.

However if you want to pass in the `name` first before the `favoriteNumber` then all you have to do is re-define your Person struct like i do below.

```javascript 
 struct person {
        string name;
        uint256 Myfavnumber;
    }
```
I really hope i did well to make you understand.

**Reply by ljb630** [reply]
Ho okay! Got it, @EngrPips thank you man :)

**Reply by EngrPips** [reply]
Awesome, I am very glad you get it buddy.
